File: src/countingValleys.js
const countingValleys = (_, s) => {
  const arr = s.split('');
  let position = 0;
  let valleys = 0;
  arr.forEach(item => {
    if (item === 'U') {
      position++;
    }
    else {
      position--;
    }

    if (item === 'U' && position === 0) {
      valleys++;
    }
  });
  return valleys;
};

module.exports = countingValleys;

File: lib/counting_valleys.rb
def countingValleys(_, s)
  arr = s.split ''
  valleys = 0
  position = 0
  arr.each do |item|
    position += 1 if item == 'U'
    position += -1 if item == 'D'
    if item == 'U' and position == 0
      valleys += 1
    end
  end
  valleys
end
